### ISSUE TITLE
Contain all page content in landmarks

### ISSUE THREAD

**Issue by atsansone** [flutter/website#10543]
From https://buganizer.corp.google.com/issues/337316892

### Observed Results

Navigating through the page, all page content should be contained by landmarks.

* [Screencast ChromeVox](https://screencast.googleplex.com/cast/NTg5NTQxNjg1MDY3Nzc2MHwzNjg2MGY5YS0xZA)
* [Screenshot](https://screenshot.googleplex.com/y2QZ9DET32eNbhC)

### Expected Results

Ensures all page content is contained by landmarks

### User Impact

Screen reader users are not able to skip to the primary content efficiently and save them time and action presses.

### Steps to Reproduce

1. Enabled ChromeVox using <kbd>Ctrl</kbd> + <kbd>Alt</kbd> + <kbd>Z</kbd>.
1. Open the following page: https://docs.flutter.dev/
1. Start the landmark navigation using "search plus ;"
1. Observe the behavior
1. Open parashu and run the automatic tool

### GAR Assessment

[Flutter Docs](https://a11ytracker.corp.google.com/products/c2fdcf18-53d2-475d-b8b0-e5dedf8cb900?e=ProductLabels)

### GAR Details

[2.8. Do you provide a bypass for headings and content that are repeated across multiple pages or within the same page?](go/GAR-web#repeated-content)

### Environment Details

Machine: Chromebook / Macbook
Browser: Chrome (123.0.6312.112 (Official Build) (64-bit))
Accessibility Tool: Chromevox / VoiceOver
Bug filling time: 15 mins.

**Comment by parlough** [comment]
I believe this is resolved by https://github.com/flutter/website/issues/10531 and other cleanup. The top nav has a `banner` role, the left and right navs have the `navigation` role, the footer has `contentinfo`, and the primary content is in `main > article` with a skip button to it.

If anyone has more concerns or issues relating to page landmarks, please feel free to reopen or open a new issue. Thanks so much!
